### ISSUE TITLE
HideTerrain bug in standalone

### ISSUE THREAD

**Issue by AvengingEagle** [TheGameCreators/GameGuruRepo#787]
My game takes place (mostly) indoors. To save drawing unnecessary polygons, I have a simple script running to hide the terrain, using the HideTerrain() command.

It works fine in test game.

In standalone however, the player gets dropped into a totally empty map containing nothing but water, whereupon he quickly begins to drown. 

I suspect the HideTerrain() command is to blame. 

I have previously experienced this bug as far back as 2018, so it's not a new issue.

AE

**Comment by synchromesh62** [comment]
Are you saving your maps to the default mapbank folder ?
This normally happens with standalones when they are saved to a custom path.

**Comment by dagored100** [comment]
I have the same problems.. or the standalone is under water.. or just black and nothing in it...
but if i check the files they are all there.
and i do not use hideterrain.. i do save to a custom path.. so i will check if that is the problem.
cheers
Dagored
Harry Wever

**Comment by dagored100** [comment]
just checked it, but save to the sdefault, also a dark level.. with nothing in it..
it starts to fail when i alter the settings in the double tab options, like dark sky , move some sliders etc..
Harry
Dagored

**Comment by synchromesh62** [comment]
Ha your map been saved to the standalone mapbank folder ?

**Comment by dagored100** [comment]
yep thats the problem, i save the map under a folder..
if i move the map.fpm to the mapbank default it works...
so it is not seeing folders under the mapbank

**Comment by AvengingEagle** [comment]
Yes, my map was saved to a subfolder within mapbank. Given that we all make so many maps and test maps, you have to have subfolders to keep things organised.

**Comment by synchromesh62** [comment]
> Yes, my map was saved to a subfolder within mapbank. Given that we all make so many maps and test maps, you have to have subfolders to keep things organised.

@AvengingEagle 
Although i agree GameGuru has always worked this way.
Did it work if you save the map directly to the mapbank folder ?

**Comment by AvengingEagle** [comment]
I moved the maps into the main mapbank folder and saved the standalone again. That appears to have  solved the issue, I am no longer dumped in water upon loading level 1. Whilst this is a workaround, I still think it's pretty draft Game Guru's mapbank can't handle subfolders. Whether or not you class this a bug is open to interpretation; I personally feel this still classes as a bug, but falls into the category of 'feature not working as expected' category, so perhaps a P2 or P3, rather than 'serious urgent-attention-required' P1 bug. As such, I still expect this to be flagged as a bug, not an enhancement.

**Comment by synchromesh62** [comment]
>  I personally feel this still classes as a bug, but falls into the category of 'feature not working as expected

Well not really … The standalone process looks for the mapfile name in the mapbank.
Maps don't have fpe's or scripts attached to tell GG to look in C:\ mymaps. Its probably the only thing that does make sense that it should be in that folder.

**Comment by AvengingEagle** [comment]
That's just it though, it's because winzones function like ifused fields on a entity, which is counter-intuitive. If I want to assign a script to an entity, I have to pick a filepath. If I want to assign a sound to a zone, I have to pick a filepath. Why would winzones be any different? It's no good to just say "it's always been this way", that's not a valid argument against improving something. And all while it's not fixed/improved, it will continue to trip users up because it doesn't work as expected.

**Comment by synchromesh62** [comment]
> it's always been this way", that's not a valid argument against improving something.
I did not say it was ? 
In fact I said its a good idea and suggested you request the feature for improvement.
Its not my fault that GG is hard coded to look in the mapbank folder for your map. I just gave you the solution to the problem you were having and tried to explain why it did it.

**Comment by AvengingEagle** [comment]
Sorry, I didn't mean to have a go, I'm just aware other people look at these threads and I didn't want them to think this has been solved because something still going awry somewhere.

Yes, I've found that if that subfolder is in mapbank, standalone will copy your maps and put them in the same subfolder (e.g. mapbank\testmap to mapbank\testmap), but by doing so some of the functionality of the levels, or the scripts they run, appears to break, like the HideTerrain() function. This is what I'm getting at when I say there's a bug somewhere. 

Try making a simple 3 level standalone with maps saved in a subfolder and each level running a simple HideTerrain() script from an always active entity. Spawn the player in a room or something. And see if in standalone you actually spawn in that room or you spawn in the sea like I seem to be doing. 

Just to be clear, yes, there does appear to be a workaround for this issue (don't save your maps in subfolders), but I'm just wondering why doing so somehow messes up certain LUA scripts.
